### ISSUE TITLE
[CodePlex] Rename doesn't catch conflicts with accessors

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#2511]
Ported from TFS WorkItem: **991467**

------------------------------

**Repro Steps:**

**Changes to this WorkItem will <u>not</u> be reflected in CodePlex and may<br>be automatically overwritten!**<br>Original repro: [http://roslyn.codeplex.com/workitem/162](http://roslyn.codeplex.com/workitem/162) to modify this bug<br>(*unless you are assigning, resolving, closing or adding private comments*).<br>**--------------------------------------------------------------------------------**<br><br>***SSL** reported:*<br><br>Source:<br><br>```C#<br>class V {<br>	int X { get; set; }<br>}<br>```<br><br>Rename `V` to `get_X`.<br><br>No conflict is shown.<br>(similarly, if you rename another member)

------------------------------

**Revisions:**

------------------------------

2) Edited By Dustin Campbell (7/22/2014 8:23:57 AM)

This causes a bunch of first-chance exceptions

System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.<br>&nbsp;&nbsp; at System.Collections.Generic.Dictionary`2.get_Item(TKey key)<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.CSharp.Rename.CSharpRenameConflictLanguageService.AddSymbolSourceSpans(List`1 conflicts, IEnumerable`1 symbols, IDictionary`2 reverseMappedLocations) in C:\Projects\roslyn\Main\Open\Workspaces\CSharp\Rename\CSharpRenameRewriterLanguageService.cs:line 876<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.CSharp.Rename.CSharpRenameConflictLanguageService.&lt;ComputeDeclarationConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\CSharp\Rename\CSharpRenameRewriterLanguageService.cs:line 732<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.&lt;AddDeclarationConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.cs:line 191<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.Session.&lt;IdentifyConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.Session.cs:line 380<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.Session.&lt;ResolveConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.Session.cs:line 135<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.AbstractEditorInlineRenameService.InlineRenameLocationSet.&lt;GetReplacementsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Closed\EditorFeatures\Core\Implementation\InlineRename\AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:line 35<br>7/22/2014 8:24:10 AM<br>Source: First chance exception: First chance exception: First chance exception: First chance exception: First chance exception: mscorlib<br>StackTrace:&nbsp;&nbsp;&nbsp; at System.Collections.Generic.Dictionary`2.get_Item(TKey key)<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.CSharp.Rename.CSharpRenameConflictLanguageService.AddSymbolSourceSpans(List`1 conflicts, IEnumerable`1 symbols, IDictionary`2 reverseMappedLocations) in C:\Projects\roslyn\Main\Open\Workspaces\CSharp\Rename\CSharpRenameRewriterLanguageService.cs:line 876<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.CSharp.Rename.CSharpRenameConflictLanguageService.&lt;ComputeDeclarationConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\CSharp\Rename\CSharpRenameRewriterLanguageService.cs:line 732<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.&lt;AddDeclarationConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.cs:line 191<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.Session.&lt;IdentifyConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.Session.cs:line 380<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Rename.ConflictEngine.ConflictResolver.Session.&lt;ResolveConflictsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Open\Workspaces\Core\Rename\ConflictEngine\ConflictResolver.Session.cs:line 135<br>--- End of stack trace from previous location where exception was thrown ---<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)<br>&nbsp;&nbsp; at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()<br>&nbsp;&nbsp; at Microsoft.CodeAnalysis.Editor.Implementation.InlineRename.AbstractEditorInlineRenameService.InlineRenameLocationSet.&lt;GetReplacementsAsync&gt;d__1.MoveNext() in C:\Projects\roslyn\Main\Closed\EditorFeatures\Core\Implementation\InlineRename\AbstractEditorInlineRenameService.InlineRenameLocationSet.cs:line 35

------------------------------

5) Edited By Kevin Pilch-Bisson (2/20/2015 12:40:18 PM)

Please port to GitHub and tag @slaks.
We could live with this scenario given the rarity of naming a type or member &quot;get_&quot;, but I don't like the exceptions that are being thrown.

**Comment by dpoeschl** [comment]
Another case from https://roslyn.codeplex.com/workitem/139:

```VB
Event B()
Dim AEventHandler
```

Renaming B to A or A to B (either direction) will silently succeed, then (correctly) show a compiler error from the new code. 

event 'A' implicitly defines 'AEventHandler', which conflicts with a member of the same name

**Comment by dpoeschl** [comment]
Note: unlike the C# case above, the VB EventHandler case does not cause any first chance exceptions.

**Comment by Pilchie** [comment]
With no internal first chance exception, I'm fine moving this to 1.1.  Given .NET naming guidelines, someone trying to name something get_X is unlikely.

**Comment by SLaks** [comment]
Note that these names are slightly more likely when implementing COM interfaces that have parameterized properties.

However, since COM doesn't support overloading, this particular conflict is still unlikely.

**Comment by CyrusNajmabadi** [comment]
Not worth doing anything here.  We've gotten no reports on this in 10 years.
